tests/test_math.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+import hcipy
+import numpy as np
+
+
+def _parameters():
+    for func in ['fft', 'ifft', 'fft2', 'ifft2', 'fftn', 'ifftn', 'rfft', 'irfft', 'rfft2', 'irfft2', 'rfftn', 'irfftn', 'hfft', 'ihfft']:
+        for dtype_in in ['float32', 'float64', 'complex64', 'complex128']:
+            real_out = func.startswith('ir') or func.startswith('h')
+
+            if dtype_in == 'float32' or dtype_in == 'complex64':
+                dtype_out = 'float32' if real_out else 'complex64'
+            else:
+                dtype_out = 'float64' if real_out else 'complex128'
+
+            if func.startswith('r') and dtype_in.startswith('complex'):
+                continue
+
+            if func.startswith('ih') and dtype_in.startswith('complex'):
+                continue
+
+            for method in ['numpy', 'scipy', 'fftw', 'mkl']:
+                if ('r' in func or 'h' in func) and method == 'mkl':
+                    # MKL doesn't implement rffts or hffts.
+                    continue
+
+                yield (func, method, dtype_in, dtype_out)
+
+@pytest.mark.parametrize('func,method,dtype_in,dtype_out', list(_parameters()))
+def test_fft_acceleration(func, method, dtype_in, dtype_out):
+    rng = np.random.default_rng(seed=0)
+
+    print(func, method, dtype_in, dtype_out)
+    if method == 'fftw':
+        pytest.importorskip('pyfftw')
+
+    if method == 'mkl':
+        pytest.importorskip('mkl_fft')
+
+    N = 128
```

Comment:
This does not test the multi-thread. Is it on purpose?

Reply:
Good catch. This should probably be a for-loop over `N`, with `N` in `[32, 128, 1024]` or something.

Reply:
`[128, 256]` should be enough to keep it minimal?

Reply:
@raphaelpclt Done by [9e2b20c](https://github.com/ehpor/hcipy/pull/213/commits/9e2b20c94c7bdb278a160356a0f3f2cfe9c53dbd).